### PR TITLE
feat-OTP-2FA

### LINKED ISSUE #19: Seller auth & Two Factor Authentication for Sellers
Description

- As a seller on the platform, I want to enhance my account security by enabling two-factor authentication (2FA). This feature will add an extra layer of security by requiring email authentication codes in addition to the standard login credentials. After a successful login and 2FA verification, sellers should be redirected to their dashboard. This feature will provide an additional level of security for sellers, ensuring their account and data are better protected.

Acceptance Criteria

- Implement a 2FA setup option in the seller's account settings.
- Allow sellers to enable or disable 2FA at any time.
- During login, if 2FA is enabled, prompt the seller to enter an email authentication code.
- Validate the entered authentication code.
- After successful 2FA verification, redirect the seller to their dashboard.
- Provide an option for sellers to resend the authentication code if not received.
- Display error messages for incorrect codes and allow retry.
- Implement appropriate error handling and responses.

Definition of Done

- The 2FA setup option is available in the seller's account settings.
- Sellers can enable or disable 2FA.
- During login, sellers are prompted for an email authentication code if 2FA is enabled.
- Authentication codes are validated correctly.
- Sellers are redirected to their dashboard after successful 2FA verification.
- The option to resend the authentication code is functional.
- Error messages for incorrect authentication codes are displayed, and retries are allowed.

### PR DESCRIPTION
**Pull request by Calebgisa72**
What does this PR do?
This PR adds and refines the OTP (One-Time Password) verification functionality for enhanced user account security during login and registration.

Description of Task to be completed
UI Implementation:

- Designed and implemented the OTP verification page with a responsive layout.
- Added input fields for users to enter the 6-digit OTP.

State Management:

- Integrated OTP input handling using React state.
- Managed OTP state updates and error handling with custom logic.

API Integration:

- Connected the OTP verification page to the backend API for OTP submission and validation.
- Added error handling to display appropriate messages for invalid or expired OTPs.
- Implemented functionality for resending OTPs and handling the API response.

Error Handling:

- Displayed user-friendly error messages for incorrect OTP or server-related issues.
- Used react-hot-toast for toast notifications to inform users about OTP status and errors.

Testing:

- Added unit tests for OTP input handling and state updates.
- Ensured the OTP component behaves correctly under various scenarios, including valid, invalid, and empty inputs.

### CODE REVIEW

**Comment by faid-terence** [comment]
@Calebgisa72 
Too many commit messages !

**Review by aimedivin — APPROVED** [review]

**Review by Ndevu12 — APPROVED** [review]

**Review by MC-Knight — APPROVED** [review]

File: src/images/iconoir_xmark-circle (2).svg
Comment (elijahladdie) [inline src/images/iconoir_xmark-circle (2).svg:1]: Can you move this image in assets folder please

File: src/pages/Authentication/GoogleLoginSuccess.tsx [hunk 1]
```diff
@@ -4,27 +4,45 @@ import { useDispatch } from 'react-redux';
...
 const GoogleLoginSuccess = () => {
   const dispatch = useDispatch<AppDispatch>();
+  const navigate = useNavigate();
+
   useEffect(() => {
     axios
       .get(`${import.meta.env.VITE_APP_API_URL}/user/login/success`, { withCredentials: true })
       .then((response) => {
-        dispatch(setCredentials(response.data.data.token));
+        console.log(response.data);
```
Comment (elijahladdie) [inline src/pages/Authentication/GoogleLoginSuccess.tsx]: remove this console

**Review by elijahladdie — CHANGES_REQUESTED** [review]

**Review by GSinseswa721 — APPROVED** [review]

**Review by faid-terence — APPROVED** [review]
Approved @Calebgisa72 !

**Pull request merged**